File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcRelationship/IfcRelDefines/IfcRelDefinesByProperties/IfcRelDefinesByProperties.vb
Public Class IfcRelDefinesByProperties Inherits IfcRelDefines
    Public RelatedObjects As SET [1:?] OF IfcObjectDefinition
    Public RelatingPropertyDefinition As IfcPropertySetDefinitionSelect

    ' === WHERE CLAUSES ===
    ' NoRelatedTypeObject : SIZEOF(QUERY(Types <* SELF\IfcRelDefinesByProperties.RelatedObjects |  'IFC4X3_DEV_73740fe4.IFCTYPEOBJECT' IN TYPEOF(Types))) = 0
End Class

File: src/main/resources/output/IFC4X3_js/Entities/IfcRoot/IfcRelationship/IfcRelDefines/IfcRelDefinesByProperties/IfcRelDefinesByProperties.js
class IfcRelDefinesByProperties extends IfcRelDefines {
    constructor() {
        /** @type {SET [1:?] OF IFCOBJECTDEFINITION} */
        this.RelatedObjects = null;
        /** @type {IFCPROPERTYSETDEFINITIONSELECT} */
        this.RelatingPropertyDefinition = null;
    }

    // === WHERE CLAUSES ===
    // NoRelatedTypeObject : SIZEOF(QUERY(Types <* SELF\IfcRelDefinesByProperties.RelatedObjects |  'IFC4X3_DEV_73740fe4.IFCTYPEOBJECT' IN TYPEOF(Types))) = 0
}
